### PR TITLE
intel and intel-parallel-studio: fix version 2015

### PR DESCRIPTION
**Pull request by mkuron**
- The `ARCH_SELECTED` parameter is not supported.
- The components have different names than in newer versions: `intel-compxe`, `intel-compilerproc`, `intel-compilerf`.
- The psxevars.sh fix doesn't apply because the directory doesn't exist.
- The installation path is different and contains version numbers.
- Also, I added version 2015.1.133 because that's what I needed.

### CODE REVIEW

**Comment by mkuron** [comment]
Hmm, looks like this is a partial duplicate of #5172.

**Comment by adamjstewart** [comment]
@svenevs

**Comment by svenevs** [comment]
@adamjstewart I don't have access to that cluster anymore.

@mkuron yours is missing the mpi stuff, but generally looks the same in spirit.  The idea Adam and I were working on was to define a single `intel_prefix` that gets set depending on e.g., `2015` vs `2017`.  It seems likely this will change again in the future, Intel seems to like to do that to their users.  Yours does some stuff about `ARCH` that I never did, though.

Your guess is as good as mine as far as what is correct.  I remember the process was very ugly, and I also remember my patches were likely not correct.  E.g. packages with intel were failing to build because it couldn't `#include <complex>`.  At that point I just gave up.

**Comment by mkuron** [comment]
My 2015 license doesn’t include MPI, so I didn’t need the MPI stuff you have. If you don’t have access to 2015 anymore, should we just go with my patch for now?

**Comment by svenevs** [comment]
I went ahead and closed the other one, but I think from a design perspective `intel_prefix` is a better approach.  This way if `2018` comes out and things are drastically different, `2018` can just make sure to set it correctly.

Where MPI is concerned, ideally that is included.  In the context of `spack`, MPI is a major component that many people will likely want to use from the intel compiler suite.  That said, it's also less likely that anybody wants to install `2015` as time goes on.

Basically, I'm washing my hands of this since I can't do any testing on it, and also generally don't care to battle intel's shenanigans.  Note that just modifying this package on its own is not complete, all of the other intel packages need updates as well (or at least from conversations with @adamjstewart, that was my understanding).
